cmd/pebble/cmd_exec.go

```diff
@@ -0,0 +1,250 @@
+// Copyright (c) 2021 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package main
+
+import (
+	"errors"
+	"os"
+	"os/signal"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/jessevdk/go-flags"
+	"golang.org/x/sys/unix"
+
+	"github.com/canonical/pebble/client"
+	"github.com/canonical/pebble/internal/logger"
+	"github.com/canonical/pebble/internal/ptyutil"
+)
+
+type cmdExec struct {
+	clientMixin
+	WorkingDir string        `long:"cwd"`
+	Env        []string      `long:"env"`
+	User       string        `long:"user"`
+	Group      string        `long:"group"`
+	Timeout    time.Duration `long:"timeout"`
+	Terminal   bool          `short:"t" long:"terminal"`
+	NoTerminal bool          `short:"T" long:"no-terminal"`
+	Positional struct {
+		Command string `positional-arg-name:"<command>" required:"1"`
+	} `positional-args:"yes"`
+}
+
+var execDescs = map[string]string{
+	"cwd":         "Working directory to run command in",
+	"env":         "Environment variable to set (in 'FOO=bar' format)",
+	"user":        "User name or ID to run command as",
+	"group":       "Group name or ID to run command as",
+	"timeout":     "Timeout after which to terminate command",
+	"terminal":    "Force pseudo-terminal allocation",
```

Comment:
`Allocate pseudo-terminal (default if stdout is a TTY).`

Reply:
Done.